src/lib.rs

```diff
@@ -24,14 +24,19 @@ const URL: &str = "http://localhost:8000";
 ///
 /// If this is ever needed outside of `get_agent` just make this
 /// function public.
-fn get_identity(account_name: impl AsRef<Path>) -> Result<BasicIdentity> {
-    let mut ident_path = dirs::config_dir().ok_or(crate::Error::MissingConfig)?;
-    ident_path.push("dfx/identity");
-    ident_path.push(account_name);
-    ident_path.push("identity.pem");
-
-    let identity = BasicIdentity::from_pem_file(ident_path)?;
-    Ok(identity)
+pub fn get_identity<'a>(account_name: impl Into<&'a str>) -> Result<BasicIdentity> {
+    let home_folder = std::env::var("HOME")?;
+    let account_name: &str = account_name.into();
+    let path_string = format!("{home_folder}/.config/dfx/identity/{account_name}/identity.pem");
+    let path = Path::new(&path_string);
+    let identity = BasicIdentity::from_pem_file(path);
+    match identity {
+        Ok(identity) => Ok(identity),
+        Err(PemError::Io(e)) if e.kind() == std::io::ErrorKind::NotFound => {
+            Err(Error::CertNotFound(path.into()))
+        }
+        Err(err) => Err(Error::from(err)),
+    }
 }
```

Comment:
Keep it as `Path` rather than make it into strings, as that's what `Path` is for.

Rather than removing `dirs` let's use `dirs.home_dir`.
Including the path in the error is a good idea.

```suggestion
pub fn get_identity(account_name: impl AsRef<Path>) -> Result<BasicIdentity> {
    let mut ident_path = dirs::home_dir().ok_or(crate::Error::MissingConfig)?;
    ident_path.push(".config");
    ident_path.push("dfx/identity");
    ident_path.push(account_name);
    ident_path.push("identity.pem");

    match BasicIdentity::from_pem_file(&ident_path) {
        Ok(identity) => Ok(identity),
        Err(PemError::Io(e)) if e.kind() == std::io::ErrorKind::NotFound => {
            Err(Error::CertNotFound(ident_path))
        }
        Err(err) => Err(Error::from(err)),
    }
}
```

Reply:
Ah, so there's a home var in dirs, ok